api/src/context.rs

```diff
@@ -121,12 +132,24 @@ impl Context {
             .get_first_txn_version()
             .map_err(|e| E::internal(e).error_code(AptosErrorCode::ReadFromStorageError))?
         {
+            let ledger_info = self
+                .get_latest_ledger_info_with_signatures()
+                .map_err(E::internal)?;
+            let (_, _, oldest_block_event) = self
+                .db
+                .get_block_info(oldest_version)
+                .map_err(|e| E::internal(e).error_code(AptosErrorCode::ReadFromStorageError))?;
```

Comment:
sorry but the new block event for the first txn version is not guaranteed to exist (mostly likely it doesn't exist).

use expose and use `lookup_event_after_version`

Reply:
Do we not prune by block?

Reply:
no.. remember we don't have blocks :trollface:

Reply:
use your new API here as well?